src/copy.jl

```diff
@@ -93,7 +113,14 @@ function copy_model(model::Model)
     reference_map = ReferenceMap(new_model, index_map)
 
     for (name, value) in object_dictionary(model)
-        new_model.obj_dict[name] = getindex.(reference_map, value)
+        if filter_constraints === nothing || 
+                (value isa ConstraintRef && filter_constraints(value)) || 
+                value isa VariableRef || 
+                eltype(value) == VariableRef
+            new_model.obj_dict[name] = getindex.(reference_map, value)
+        elseif eltype(value) == ConstraintRef && any(filter_constraints.(value))
+            new_model.obj_dict[name] = getindex.(reference_map, filter(filter_constraints, value))
+        end
```

Comment:
We generally try not to use `isa` like this. How about something like
```Julia
_copy_object(filter, value) = true
_copy_object(filter::Function, value::ConstraintRef) = filter(value)
function _copy_object(filter::Function, value::AbstractArray{<:ConstraintRef})
    return all(filter.(value))
end

for (name, value) in object_dictionary(model)
    if _copy_object(filter_constraints, value)
        new_model.obj_dict[name] = getindex.(reference_map, value)
    end
end
```

What does the `elseif` clause do when we only want some of the constraints? That seems like an issue. We probably want to just skip copying the name into `obj_dict` if only some of the container is copied.

Reply:
That reasoning would lead to problems when copying just a conflict: maybe only a few constraints of the container are in the conflict, and only those should be copied. I'll propose a new iteration on the code soon.

Reply:
To clarify: we should copy on the constraints in the conflict, but maybe not set the name in `.obj_dict`.

Reply:
I think I implemented your suggestion: (1) use functions instead of `isa` and (2) only copy into the dictionary the objects that are completely copied.